packages/tools/client-debugger/client-debugger/src/data-visualization/DataVisualization.ts

```diff
@@ -344,8 +346,9 @@ export class VisualizerNode extends TypedEventEmitter<DataVisualizerEvents> impl
 	 * {@link SharedObjectListenerEvents | "update"} event.
 	 */
 	private emitVisualUpdate(): void {
-		const visualTree = this.render();
-		this.emit("update", visualTree);
+		this.render().then((visualTree: FluidObjectNode) => {
+			this.emit("update", visualTree);
+		}, console.error);
```

Comment:
Found the Promise 😄 . However, did we explore the option of making `emitVisualUpdate` async and awaiting the call to `this.render()`? There's a few call sites for it, so maybe knowing that the actual emit of the "update" event is an async thing would help prevent other bugs.

E.g. event listeners that call this and then `return true;` to indicate that they handled the message (`onOpHandler` on line 292). It feels like that should await the call to `emitVisualUpdate` before returning.

Reply:
That's a fair point. It puts the burden on the callers to handle the promise, but that might be better (and there are only 2 callers).

Reply:
Added async / await for `onOpHandler`.